src/main/java/SYSC6/Project/RoleType.java

```diff
@@ -1,8 +1,43 @@
 package SYSC6.Project;
 
+import javax.management.relation.Role;
+import java.util.Locale;
+import java.util.Optional;
+
 public enum RoleType {
-    NO_ROLE,
-    FREE_USER,
-    PAID_USER,
-    ADMIN
+    NO_ROLE (1, "no_role"),
+    FREE_USER (2, "free_user"),
+    PAID_USER (1, "paid_user"),
+    ADMIN (3, "admin");
```

Comment:
The index parameter here is used to indicate the ordinal of the enum this role may upgrade to.
Its a bit silly, and we have an endpoint where we can just specify as well, but I implemented this system regardless